examples/magento-graphcms/pages/[...url].tsx

```diff
@@ -111,46 +99,17 @@ function CategoryPage(props: CategoryProps) {
         <>
           <CategoryDescription description={category.description} />
           <CategoryChildren params={params}>{category.children}</CategoryChildren>
-          <StickyBelowHeader>
-            {import.meta.graphCommerce.productFiltersPro ? (
-              <ProductFiltersPro params={params}>
-                <ProductListFiltersContainer>
-                  <ProductFiltersProFilterChips
-                    {...filters}
-                    appliedAggregations={products.aggregations}
-                    filterTypes={filterTypes}
-                  />
-                  <ProductFiltersProSortChip {...products} />
-                  <ProductFiltersProLimitChip />
-                  <ProductFiltersProAllFiltersChip
-                    {...products}
-                    {...filters}
-                    appliedAggregations={products.aggregations}
-                    filterTypes={filterTypes}
-                  />
-                </ProductListFiltersContainer>
-              </ProductFiltersPro>
-            ) : (
-              <ProductListParamsProvider value={params}>
-                <ProductListFiltersContainer>
-                  <ProductListSort
-                    sort_fields={products?.sort_fields}
-                    total_count={products?.total_count}
-                  />
-                  <ProductListFilters {...filters} filterTypes={filterTypes} />
-                </ProductListFiltersContainer>
-              </ProductListParamsProvider>
-            )}
-          </StickyBelowHeader>
-          <Container maxWidth={false}>
-            <ProductListCount total_count={products?.total_count} />
-            <ProductListItems
-              title={category.name ?? ''}
-              items={products?.items}
-              loadingEager={1}
-            />
-            <ProductListPagination page_info={products?.page_info} params={params} />
-          </Container>
+          <FilterLayout
+            mode={matches ? 'sidebar' : 'default'}
+            ProductListItems={ProductListItems}
+            ProductListCount={ProductListCount}
+            ProductListPagination={ProductListPagination}
+            products={products}
+            filters={filters}
+            params={params}
+            filterTypes={filterTypes}
+            category={category}
+          />
```

Comment:
Can we first try to build this without creating this abstraction?

Might be interesting to create a FilterLayout which is _only_ a presentation layout? Also the Layout would only apply for the ProductFiltersPro of course.